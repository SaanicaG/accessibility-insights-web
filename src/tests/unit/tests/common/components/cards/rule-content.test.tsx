// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { RuleContent, RuleContentProps } from 'common/components/cards/rule-content';
import * as React from 'react';
import {                                              
    mockReactComponents,                    
} from 'tests/unit/mock-helpers/mock-module-helpers';   
import {InstanceDetailsGroup} from 'common/components/cards/instance-details-group';
jest.mock('common/components/cards/instance-details-group');  

describe('RuleContent', () => {
    mockReactComponents([InstanceDetailsGroup]);
    it('renders', () => {
        const props = {
            rule: {
                id: 'test-id',
            },
        } as RuleContentProps;

        const renderResult = render(<RuleContent {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
