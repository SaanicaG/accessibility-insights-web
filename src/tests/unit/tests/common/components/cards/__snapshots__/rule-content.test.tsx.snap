// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleContent renders 1`] = `
<DocumentFragment>
  <mock-instancedetailsgroup
    rule="[object Object]"
  />
</DocumentFragment>
`;

exports[`RuleContent renders: InstanceDetailsGroup props 1`] = `
{
  "rule": {
    "id": "test-id",
  },
}
`;
